src/internal/_mergeWith.js

```diff
@@ -0,0 +1,59 @@
+var curryN = require('../curryN');
+
+
+/**
+ * Merges objects `l` and `r`. If any keys exist in both objects, `fn` is
+ * applied to the key and the values of each object to determine the value to
+ * use. If `undefined` is returned, the key is excluded from the merged object.
+ * An object consisting of keys that only exist in `l` and not `r` are given to
+ * `fnL`, the result of which is merged back into the object to be returned.
+ * Similarly, keys that exist in `r` and not `l` will be given to `fnR` whose
+ * result will also be merged in. The `fnL` and `fnR` functions provide the
+ * ability to filter a result, e.g. to produce the intersection or difference
+ * of two objects using `R.always({})`.
+ */
+module.exports = curryN(5, function _mergeWith(fn, fnL, fnR, l, r) {
+  var lNotR = {};
+  var rNotL = {};
+  var skipped = {};
+  var result = {};
+  var k, v, lMerged, rMerged;
+
+  for (k in l) {
+    if (l.hasOwnProperty(k)) {
+      if (r.hasOwnProperty(k)) {
```

Comment:
We should use `Object.prototype.hasOwnProperty.call` (or `_has`) to handle objects with own `hasOwnProperty` properties. ;)